front/pages/poke/[wId]/data_sources/[name]/index.tsx

```diff
@@ -311,6 +326,34 @@ const DataSourcePage = ({
     }
   });
 
+  const { submit: handleWhiteListedChannelPatternsChange } = useSubmitFunction(
```

Comment:
Any reason to know move this code next to the form?

Reply:
Only because I saw all the `*Toggle` functions defined here. But can actually move it to the form definition if we don't want to overload this file

Reply:
I'm not strongly held on this, but given that you created a dedicated file, I'd love to see all the logic living in one place.